### PR TITLE
feat: Default H2 schema through config instead of copy-paste 

### PR DESCRIPTION
**Pull request by johanandren**
References #935

Added dialect in test names as well for easier tracking in case test failures are related to some specific dialect

### CODE REVIEW

File: docs/src/main/paradox/r2dbc.md [hunk 1]
```diff
@@ -56,11 +56,15 @@ YugaByte
 :  @@snip [YugaByte Schema](/akka-projection-r2dbc/ddl-scripts/create_tables_yugabyte.sql)
 
 H2
-:  @@snip [H2 Schema](/akka-projection-r2dbc/src/it/resources/application-h2.conf) { #schema }
+:  @@snip [H2 Schema](/akka-projection-r2dbc/src/main/resources/h2-default-projection-schema.conf) { #schema }
```
Comment (patriknw) [inline docs/src/main/paradox/r2dbc.md:59]: the previous snippet showed this part:
```
akka.persistence.r2dbc.connection-factory {
  additional-init = """
```

That was quite useful, could we somehow make it show? Does hcon support `/*` comments?

```
akka.projection.r2dbc.default-h2-schema =
/*
// #schema
akka.persistence.r2dbc.connection-factory {
  additional-init = """
// #schema
*/
  """
// #schema
  CREATE TABLE
```

**Review by patriknw — COMMENTED** [review]
good to have this in a plain config file that can be included

File: docs/src/main/paradox/r2dbc.md (hunk 1 above)
Comment (johanandren) [reply docs/src/main/paradox/r2dbc.md:59]: No support for multi line comments in hocon, but I figured out some other trixery to do it.

File: akka-projection-r2dbc/src/main/resources/h2-default-projection-schema.conf [hunk 2]
```diff
@@ -0,0 +1,55 @@
+# Default schema for H2 projections
+
+# Ignore this section, just here for easy copy paste schema in docs
+# Note: keep the weird indentation
```
Comment (patriknw) [inline akka-projection-r2dbc/src/main/resources/h2-default-projection-schema.conf:4]: 👍 😅

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**